test/strict/classComputedPropertyName.js

```diff
@@ -0,0 +1,245 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+    {
+        name: "Assigning an undeclared variable in a class' computed property name",
+        body: function () {
+            assert.throws(
+                function () {
+                    class C {
+                        [f = 5]() { }

```

Comment:
The execution of 

```
function () {
     class C {
          [f = 5]() { }
     }                
}
```
and
```
function () {
"use strict";
     class C {
          [f = 5]() { }
     }                
}
```
is the same

Reply:
Might be a good idea to add a test case anyway, just in case some future semantics change breaks or changes this behavior.

---
In reply to: [216857560](https://github.com/Microsoft/ChakraCore/pull/5689#discussion_r216857560) [](ancestors = 216857560)

Reply:
84292eb